app/src/main/java/org/jellyfin/androidtv/util/profile/ProfileHelper.kt

```diff
@@ -15,6 +15,34 @@ import timber.log.Timber
 object ProfileHelper {
 	private val MediaTest by lazy { MediaCodecCapabilitiesTest() }
 
+	private val supports4K by lazy {
+		val (maxWidth, maxHeight) = MediaTest.getMaxResolution()
+		maxWidth >= 3840 && maxHeight >= 2160
+	}
+
+	val max1080pCodecProfile by lazy {
+		CodecProfile().apply {
+			type = CodecType.Video
+			conditions = if (supports4K) {
+				// No restriction if 4K is supported
+				arrayOf()
```

Comment:
We should never add empty codec profiles

Reply:
Added to support up to 4k resolutions instead.